### ISSUE TITLE
Joint Learning-based KoBERT   for Emotion Recognition in Korean 

### ISSUE THREAD

**Issue by hs-sugi** [One-Day-One-Paper/One-Day-One-Paper#167]
# 한국어  감정  분석을  위한 합동  학습  기반  KoBERT  모델 
- 한국정보과학회 2020 한국소프트웨어종합학술대회 논문집
- 이영준 (한국과학기술원) ,  최호진 (한국과학기술원)

### Abstract
- 한국어 감정 사전: 트위터 데이터셋  이용
  - Weighted Tweet Frequency (TwF) 방법 & Thesaurus + Translation (TT) 방법으로 구축
  - Ekman의  감정  모델에  기반해  총  6가지 감정  목록  포함
- 단어 단위  감정  분류  태스크를  위해  입력  문장에  대한  정답 라벨  구성
  - 어절  단위로  토큰화된 문장을  입력으로  받고,  한국어  감정  사전에  속한  단어가 입력 문장 내 존재한다면 해당 위치의 정답 라벨을 [EMO]  로  설정; 그렇지  않다면,  해당  위치의  정답 라벨을  [NONEMO]  로  구성
- 문장  내  문맥을  고려한  문장과  토큰의 의미 정보를 만들기 위해 KoBERT 모델 사용
- 문장  단위  감정  분류 
  - KoBERT 모델이 어절 단위로 토큰화된 문장을 입력으로 받고, 모델에서 출력되는 첫 번째 토큰인 [CLS]  토큰의  hidden  state인  𝑜을  통해  입력  문장의 함축적  의미  정보를  파악한다.  다음으로,  입력  문장의 감정은  아래의  수식을  통해  예측한다.
![image](https://user-images.githubusercontent.com/40443049/167289216-62a78bb7-0482-4b94-8508-e4a8b5f90eba.png)

- 단어  단위  감정  분류 
  - KoBERT  모델이  토큰화된  문장을 입력으로 받으면, 모델에서  출력되는  토큰들  중 입력 문장의  위치에  상응하는  출력  토큰들인  ℎ1, ℎ3, ..., ℎ𝑇𝑇 를 통해 단어 단위의 함축적 의미 정보를 파악한다. 다음으로, 입력 문장의 각 단어들이 감정 단어인지 아닌지는  아래의  수식을  통해  예측한다.
![image](https://user-images.githubusercontent.com/40443049/167289242-7a8185da-28cc-4087-ae46-037500729dd7.png)

- 문장  및  단어  단위  감정  합동  학습
  - 학습  데이터에  대해  아래 조건부  확률을  최대화  되는  방향으로 학습이  이루어지고,  실제  구현에서는  cross entropy loss 값이 최소화 되는 방향으로 end-to-end 방식으로 학습이  진행된다.  또한,  KoBERT  모델의  구조상  해당 조건부  확률을 독립적으로  계산한다.
![image](https://user-images.githubusercontent.com/40443049/167289266-528b1ebb-8e9e-4517-90d0-e5e557cc9cdc.png)

- SNS 및 온라인 댓글 크롤링한 데이터 셋에 대해 성능 실험 결과
![image](https://user-images.githubusercontent.com/40443049/167289299-1b46106f-53f0-45b0-a6e9-49c5e6a51e65.png)
-기존 KoBERT  를  fine-tuning  시킨  모델보단  성능  향상
